### PR TITLE
[235] Add staging AKS env

### PR DESCRIPTION
**Pull request by smneal**
### Context
Add staging environment to the AKS platform

### Changes proposed in this pull request

- Add staging AKS Terraform config with some minor other improvements
- Add staging AKS Ruby config
- Add staging AKS DNS config
- Add staging AKS to GitHub actions with some minor other environments
- Add staging AKS to Makefile and create associated config
- Remove env_config which was effectively redundant
- Add additional Postgres extension which was missing

### Guidance to review

- [x]  [Deployment to staging_aks complete](https://github.com/DFE-Digital/publish-teacher-training/actions/runs/5321643489/jobs/9637003255)
- [x]  [Restore AKS database complete](https://github.com/DFE-Digital/publish-teacher-training/actions/runs/5320779535/jobs/9635001774)
- [x]  Find accessible at: https://staging2.find-postgraduate-teacher-training.service.gov.uk/
- [x]  Publish accessible at: https://staging2.publish-teacher-training-courses.service.gov.uk/
- [x]  Publish API accessible at: https://staging2.api.publish-teacher-training-courses.service.gov.uk/docs/#about
- [ ]  Validation Trello [card](https://trello.com/c/0IcaV60o)

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

**Comment by smneal** [comment]
[FindPub - Validate staging env on AKS](https://trello.com/c/upp7aCi1/436-findpub-validate-staging-env-on-aks)

File: .github/actions/deploy_v2/action.yml [hunk 1]
```diff
@@ -45,16 +45,13 @@ runs:
           echo "deploy_url=https://publish-review-${APP_NAME}.test.teacherservices.cloud" >> $GITHUB_OUTPUT
           echo "DEPLOY_REF=${{ github.head_ref }}" >> $GITHUB_ENV
         else
+          app_environment=$(jq -r '.app_environment' ${tf_vars_file})
           echo "DEPLOY_REF=${{ github.ref }}" >> $GITHUB_ENV
           hostname=$(jq -r '.additional_hostnames[0]' ${tf_vars_file})
-          if [ ${{ inputs.environment }} == "production" ]; then
-            echo "deploy_url=https://api.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
+          if [ ${app_environment} == "production" ]; then
+            echo "deploy_url=https://api2.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (smneal) [reply .github/actions/deploy_v2/action.yml:52]: Good question. It's the same as the PaaS deploy action 😄

File: .github/actions/deploy_v2/action.yml [hunk 2]
```diff
@@ -45,16 +45,13 @@ runs:
...
           else
-            if [[ $cluster == 'production' ]]; then
-              echo "deploy_url=https://publish-${aks_app_environment}.teacherservices.cloud" >> $GITHUB_OUTPUT
-            else
-              echo "deploy_url=https://publish-${aks_app_environment}.${cluster}.teacherservices.cloud" >> $GITHUB_OUTPUT
-            fi
+            echo "deploy_url=https://${app_environment}2.api.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (smneal) [reply .github/actions/deploy_v2/action.yml:54]: See previous

File: terraform/aks/workspace_variables/staging_aks.tfvars.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+  "app_environment": "staging",
+  "key_vault_name": "s189t01-ptt-stg-kv",
+  "key_vault_app_secret_name": "PUBLISH-APP-SECRETS-STAGING",
+  "key_vault_infra_secret_name": "BAT-INFRA-SECRETS-STAGING",
+  "cluster": "test",
+  "namespace": "bat-qa",
```
Comment (RMcVelia) [inline terraform/aks/workspace_variables/staging_aks.tfvars.json:7]: bat-staging

File: .github/actions/restore/action.yml [hunk 4]
```diff
@@ -49,26 +50,22 @@ runs:
...
+          echo "app_name=publish-review-${{ inputs.pr-number }}" >> $GITHUB_ENV
         else
           echo "Set the app name to the environent name"
-          echo "app_name=publish-${modified_environment}" >> $GITHUB_ENV
+          echo "app_name=publish-${{ env.app_environment }}" >> $GITHUB_ENV
         fi
 
     - name: Install kubectl
-      if: ${{ inputs.environment != 'review_aks' }}
+      if: ${{ env.app_environment != 'review' }}
```
Comment (RMcVelia) [inline .github/actions/restore/action.yml:62]: what happens to the workflow if it's a review app?

File: terraform/aks/workspace_variables/staging_aks.tfvars.json (hunk 3 above)
Comment (smneal) [reply terraform/aks/workspace_variables/staging_aks.tfvars.json:7]: As per discussion

File: .github/actions/restore/action.yml (hunk 4 above)
Comment (smneal) [reply .github/actions/restore/action.yml:62]: As per discussion

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**